### ISSUE TITLE
运行run.py的过程为什么GPU没有使用多少，迭代周期过长？

### ISSUE THREAD

**Issue by ly3106** [WeijingShi/Point-GNN#92]
Weijing：
您好！
&emsp;&emsp;当我按照您README中的说明使用run.py运行处理点云时，发现迭代周期很长，但是GPU并没有怎么时使用到，如下图所示。我看到您run.py中有设置GPU自动分配现存。所以是我打开方式不对，还是程序就没有过多使用GPU计算？我的电脑配置是Intel® Core™ i5-10300H CPU @ 2.50GHz × 8 处理器和NVIDIA GeForce RTX 2070 with Max-Q Design/PCIe/SSE2显卡
![2022-04-07 172738](https://user-images.githubusercontent.com/42125674/162169760-08c6a33a-eba6-4265-aa0a-02d6a6d9458d.jpg)

**Comment by felix0903** [comment]
您好，请问您这个问题解决了吗？

**Comment by ly3106** [comment]
> 

解决了，时间过久了，记不太清了，可能是当时tensorflow的cuda和cudnn依赖没有安装成功，导致它不能使用GPU只能使用CPU，所以才出现上面的问题。在RTX3090显卡上运行，正常情况下的GPU显存占用将近9000M，显卡利用率33%左右。建议使用anaconda安装，切记一定要把conda和pip的源都配置到境内，然后安装tensorflow环境。另外我写了一个博客[Point-GNN README批注](https://blog.csdn.net/ly869915532/article/details/123972573)希望对你有帮助。关于Point-GNN的详细运行时的一些表现，我的文章中有更为详细的参数，这是它的[源码](https://github.com/ly3106/TriNN)，但是文章还没有出版，近期会出版，等出版之后文章和源码都会开源，你可以参考。

**Comment by felix0903** [comment]
感谢您的分享，这对我目前的学习有莫大的帮助

**Comment by felix0903** [comment]
您好，我的显卡也是3090显卡，源码中的tensorflow1.x版本太低了，在环境配置过程中总是遇到tensorflow版本兼容的问题。1.如果方便的话能否分享一下你的环境版本号（Tensorflow、CUDA、CUDNN）？2.是否需要自己把源码重构到tensorflow2.x版本呢？

**Comment by ly3106** [comment]
## 解决方法
1. 更换TensorFlow版本为2.0及以后的版本， TensorFlow官方1.15.2、1.15.3、1.15.4、1.15.5 都不行，它们只支持cuda 10.0，而cuda 10.0不支持30系列及之后的网卡
2. 更换为Nvidia维护的TensorFlow 1.15.x
3. 使用Docker，参考《[RTX3090运行Tensorflow1.15（CUDA 11.1） Docker、TF1.15测试环境](https://zhuanlan.zhihu.com/p/341969571)》

我使用的是第二种方法，重构源码带来的未知太多（如果你想重构代码如果可以欢迎分享源码），无法进行控制变量的比较，下面是我的配置：

## 双3090显卡配置
Name|Version|Build|Channel
|:--|:--|--:|:--|
_libgcc_mutex|0.1|main|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
_openmp_mutex|5.1|1_gnu|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
absl-py|1.4.0|pypi_0|pypi
addict|2.4.0|pypi_0|pypi
astor|0.8.1|pypi_0|pypi
asttokens|2.2.1|pypi_0|pypi
astunparse|1.6.3|pypi_0|pypi
attrs|23.1.0|pypi_0|pypi
backcall|0.2.0|pypi_0|pypi
blinker|1.6.2|pypi_0|pypi
ca-certificates|2023.01.10|h06a4308_0|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
cachetools|5.3.0|pypi_0|pypi
certifi|2022.12.7|pypi_0|pypi
charset-normalizer|3.1.0|pypi_0|pypi
click|8.1.3|pypi_0|pypi
comm|0.1.3|pypi_0|pypi
configargparse|1.5.3|pypi_0|pypi
contourpy|1.0.7|pypi_0|pypi
cycler|0.11.0|pypi_0|pypi
dash|2.9.3|pypi_0|pypi
dash-core-components|2.0.0|pypi_0|pypi
dash-html-components|2.0.0|pypi_0|pypi
dash-table|5.0.0|pypi_0|pypi
debugpy|1.6.7|pypi_0|pypi
decorator|5.1.1|pypi_0|pypi
executing|1.2.0|pypi_0|pypi
fastjsonschema|2.16.3|pypi_0|pypi
flask|2.3.2|pypi_0|pypi
fonttools|4.39.3|pypi_0|pypi
gast|0.3.3|pypi_0|pypi
google-auth|2.17.3|pypi_0|pypi
google-auth-oauthlib|1.0.0|pypi_0|pypi
google-pasta|0.2.0|pypi_0|pypi
grpcio|1.54.0|pypi_0|pypi
h5py|2.10.0|pypi_0|pypi
idna|3.4|pypi_0|pypi
importlib-metadata|6.6.0|pypi_0|pypi
importlib-resources|5.12.0|pypi_0|pypi
ipykernel|6.22.0|pypi_0|pypi
ipython|8.12.2|pypi_0|pypi
ipywidgets|8.0.6|pypi_0|pypi
itsdangerous|2.1.2|pypi_0|pypi
jedi|0.18.2|pypi_0|pypi
jinja2|3.1.2|pypi_0|pypi
joblib|1.2.0|pypi_0|pypi
jsonschema|4.17.3|pypi_0|pypi
jupyter-client|8.2.0|pypi_0|pypi
jupyter-core|5.3.0|pypi_0|pypi
jupyterlab-widgets|3.0.7|pypi_0|pypi
keras-applications|1.0.8|pypi_0|pypi
keras-preprocessing|1.1.2|pypi_0|pypi
kiwisolver|1.4.4|pypi_0|pypi
ld_impl_linux-64|2.38|h1181459_1|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libffi|3.4.2|h6a678d5_6|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libgcc-ng|11.2.0|h1234567_1|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libgomp|11.2.0|h1234567_1|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
libstdcxx-ng|11.2.0|h1234567_1|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
markdown|3.4.3|pypi_0|pypi
markupsafe|2.1.2|pypi_0|pypi
matplotlib|3.7.1|pypi_0|pypi
matplotlib-inline|0.1.6|pypi_0|pypi
nbformat|5.7.0|pypi_0|pypi
ncurses|6.4|h6a678d5_0|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
nest-asyncio|1.5.6|pypi_0|pypi
numpy|1.23.5|pypi_0|pypi
nvidia-cublas-cu12|12.1.3.1|pypi_0|pypi
nvidia-cuda-cupti-cu12|12.1.105|pypi_0|pypi
nvidia-cuda-nvcc-cu12|12.1.105|pypi_0|pypi
nvidia-cuda-runtime-cu12|12.1.105|pypi_0|pypi
nvidia-cudnn-cu12|8.9.1.23|pypi_0|pypi
nvidia-cufft-cu12|11.0.2.54|pypi_0|pypi
nvidia-curand-cu12|10.3.2.106|pypi_0|pypi
nvidia-cusolver-cu12|11.4.5.107|pypi_0|pypi
nvidia-cusparse-cu12|12.1.0.106|pypi_0|pypi
nvidia-dali-cuda110|1.23.0|pypi_0|pypi
nvidia-dali-nvtf-plugin|1.23.0+nv23.03|pypi_0|pypi
nvidia-nccl-cu12|2.18.1|pypi_0|pypi
nvidia-nvjitlink-cu12|12.1.105|pypi_0|pypi
nvidia-pyindex|1.0.9|pypi_0|pypi
nvidia-tensorflow|1.15.5+nv23.03|pypi_0|pypi
oauthlib|3.2.2|pypi_0|pypi
open3d|0.17.0|pypi_0|pypi
opencv-python|4.7.0.72|pypi_0|pypi
openssl|1.1.1t|h7f8727e_0|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
opt-einsum|3.3.0|pypi_0|pypi
packaging|23.1|pypi_0|pypi
pandas|2.0.1|pypi_0|pypi
parso|0.8.3|pypi_0|pypi
pexpect|4.8.0|pypi_0|pypi
pickleshare|0.7.5|pypi_0|pypi
pillow|9.5.0|pypi_0|pypi
pip|23.0.1|py38h06a4308_0|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
pkgutil-resolve-name|1.3.10|pypi_0|pypi
platformdirs|3.5.0|pypi_0|pypi
plotly|5.14.1|pypi_0|pypi
prompt-toolkit|3.0.38|pypi_0|pypi
protobuf|3.20.3|pypi_0|pypi
psutil|5.9.5|pypi_0|pypi
ptyprocess|0.7.0|pypi_0|pypi
pure-eval|0.2.2|pypi_0|pypi
pyasn1|0.5.0|pypi_0|pypi
pyasn1-modules|0.3.0|pypi_0|pypi
pygments|2.15.1|pypi_0|pypi
pyparsing|3.0.9|pypi_0|pypi
pyquaternion|0.9.9|pypi_0|pypi
pyrsistent|0.19.3|pypi_0|pypi
python|3.8.16|h7a1cb2a_3|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
python-dateutil|2.8.2|pypi_0|pypi
pytz|2023.3|pypi_0|pypi
pyyaml|6|pypi_0|pypi
pyzmq|25.0.2|pypi_0|pypi
readline|8.2|h5eee18b_0|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
requests|2.30.0|pypi_0|pypi
requests-oauthlib|1.3.1|pypi_0|pypi
rsa|4.9|pypi_0|pypi
scikit-learn|1.2.2|pypi_0|pypi
scipy|1.10.1|pypi_0|pypi
setuptools|66.0.0|py38h06a4308_0|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
shapely|2.0.1|pypi_0|pypi
six|1.16.0|pypi_0|pypi
sqlite|3.41.2|h5eee18b_0|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
stack-data|0.6.2|pypi_0|pypi
tenacity|8.2.2|pypi_0|pypi
tensorboard|2.13.0|pypi_0|pypi
tensorboard-data-server|0.7.0|pypi_0|pypi
tensorflow-estimator|1.15.1|pypi_0|pypi
tensorrt|8.6.1|pypi_0|pypi
tensorrt-bindings|8.6.1|pypi_0|pypi
tensorrt-libs|8.6.1|pypi_0|pypi
termcolor|2.3.0|pypi_0|pypi
threadpoolctl|3.1.0|pypi_0|pypi
tk|8.6.12|h1ccaba5_0|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
tornado|6.3.1|pypi_0|pypi
tqdm|4.65.0|pypi_0|pypi
traitlets|5.9.0|pypi_0|pypi
typing-extensions|4.5.0|pypi_0|pypi
tzdata|2023.3|pypi_0|pypi
urllib3|2.0.2|pypi_0|pypi
wcwidth|0.2.6|pypi_0|pypi
werkzeug|2.3.3|pypi_0|pypi
wheel|0.38.4|py38h06a4308_0|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
widgetsnbextension|4.0.7|pypi_0|pypi
wrapt|1.15.0|pypi_0|pypi
xz|5.4.2|h5eee18b_0|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
zipp|3.15.0|pypi_0|pypi
zlib|1.2.13|h5eee18b_0|http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
